custom/icds_reports/sqldata/exports/dashboard_usage.py

```diff
@@ -58,206 +56,138 @@ def get_role_from_username(self, username):
-                location_type_filter[user_location_type_name] = user_location.get_id
-            else:
-                user_location_type_name = None
-            for test_location in self.location_test_fields:
-                location_type_filter[test_location] = 0
 
-            all_awc_locations = AwcLocation.objects.filter(**location_type_filter).values(*self.required_fields)
-            # converting the result set to matrix to fetch ancestors for a given location
-            location_matrix, location_ids =\
-                self.convert_rs_to_matrix(all_awc_locations, user_location_type_name)
+        location_type_filter = {
+            'aggregation_level': 5
+        }
+        if not self.national_user:
+            user_location = self.user.get_sql_location(self.domain)
+            user_location_type_name = \
+                self.get_location_id_string_from_location_type(user_location.location_type_name)
+            location_type_filter[user_location_type_name] = user_location.get_id
+        else:
+            user_location_type_name = None
+        for test_location in self.location_test_fields:
+            location_type_filter[test_location] = 0
+
+        all_awc_locations = AwcLocation.objects.filter(**location_type_filter).values(*self.required_fields)
+
+        self.populate_location_id_vs_name_mapping(all_awc_locations, user_location_type_name)
 
-            users = self.get_users_by_location(location_ids)
+        date = (datetime.datetime.now() - relativedelta(days=1)).date()
+        usage_data = None
 
-            dashboard_uname_rx = re.compile(r'^\d*\.[a-zA-Z]*@.*')
+        dashboard_filters = {}
 
-            usernames = [user['username'] for user in users if dashboard_uname_rx.match(user['username'])]
+        if not self.national_user:
+            # retrieving the user_level from logged in user location type
+            user_level = self.user_levels.index(user_location_type_name.replace('_id', '')) + 1
+            dashboard_filters['user_level__gt'] = user_level
+
+            dashboard_filters[user_location_type_name] = user_location.get_id
 
-            records = list(ICDSAuditEntryRecord.objects.filter(url='/a/{}/icds_export_indicator'
-                                                               .format(self.domain),
-                                                               username__in=usernames,
-                                                               time_of_use__gte=start_date,
-                                                               time_of_use__lt=end_date)
-                           .annotate(indicator=KeyTextTransform('indicator', 'post_data')).values('indicator',
-                                                                                                  'username')
-                           .annotate(count=Count('indicator')).order_by('username', 'indicator'))
+        # keep the record in searched - current - month
+        while usage_data is None:
```

Comment:
I hope you tested this that it is going back in days if today's table is not present